### PR TITLE
Update Scheduling page functionality

### PR DESCRIPTION
**Pull request by smnkrt**
- adds functionality for filtering tabs
- adds styling for selecting active tab and users
- adds sorting for users in tabs
- move fltering/sorting logic to from frontend to backend

### CODE REVIEW

File: app/react/services/scheduling/Filters.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import ProjectStore from '../../stores/ProjectStore'
```
Comment (azranel) [inline app/react/services/scheduling/Filters.js]: If you don't use you can delete it :smile:

File: app/services/scheduling/column_sets_builder.rb [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
+    private
+
+    USER = 'User'.freeze
+    ROLE = 'Role'.freeze
+    CURRENT = 'Current project'.freeze
+    NEXT = 'Next project'.freeze
+    BOOKED = 'Booked'.freeze
+    NOTES = 'Notes'.freeze
+
+    COLUMNS = {
```
Comment (azranel) [inline app/services/scheduling/column_sets_builder.rb]: :+1:

File: app/react/services/scheduling/Filters.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export default class Filters {
+  static selectUsers(users, store) {
+    return users.filter(user => store.userIds.indexOf(user.id) > -1);
```
Comment (azranel) [inline app/react/services/scheduling/Filters.js]: You could use lodash here (I think it's not in package.json so you would need to install it).

```js
return _.where(users, { id: user.id});
```

File: app/serializers/user_scheduling_serializer.rb [hunk 4]
```diff
@@ -12,4 +12,12 @@ def primary_role
     role = object.primary_roles[0]
     { name: role.try(:name), id: role.try(:id) }
   end
+
+  def rotation_class
+    membership_start_date = object.longest_current_membership ? object.longest_current_membership.starts_at : Date.today
+    months_in_project = (Date.today - membership_start_date)/30.to_f
```
Comment (azranel) [inline app/serializers/user_scheduling_serializer.rb]: :+1: It increase readability :smile:

File: app/services/scheduling/column_sets_builder.rb (hunk 2 above)
Comment (smnkrt) [inline app/services/scheduling/column_sets_builder.rb]: yep, i've set freeze on all of them

File: app/serializers/user_scheduling_serializer.rb [hunk 5]
```diff
@@ -12,4 +12,12 @@ def primary_role
     role = object.primary_roles[0]
     { name: role.try(:name), id: role.try(:id) }
   end
+
+  def rotation_class
+    membership_start_date = object.longest_current_membership ? object.longest_current_membership.starts_at : Date.today
```
Comment (azranel) [inline app/serializers/user_scheduling_serializer.rb]: That's one long, could you break it? :smile: I think that even `if` statement would be good

File: app/repositories/scheduled_users_repository.rb [hunk 6]
```diff
@@ -1,60 +1,72 @@
...
 
   def in_commercial_projects_with_due_date
-    not_booked_billable_users.without_scheduled_commercial_memberships
+    @in_commercial_projects_with_due_date ||= not_booked_billable_users.without_scheduled_commercial_memberships
       .joins(current_memberships: [:project])
       .where("(projects.internal = 'f') AND (memberships.ends_at > :now OR projects.end_at > :now)",
         now: 1.day.ago)
       .merge(Project.active.commercial.started.not_maintenance)
-      .order('COALESCE(memberships.ends_at, projects.end_at)')
+      .order('COALESCE(memberships.starts_at, projects.starts_at)')
```
Comment (azranel) [inline app/repositories/scheduled_users_repository.rb]: Those lines are too big to align them.

File: app/react/stores/SchedulingFilterStore.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import alt from '../alt';
 import SchedulingFilterActions from '../actions/SchedulingFilterActions';
+import * as FilterTabs from '../constants/scheduling/FilterTabs';
```
Comment (azranel) [inline app/react/stores/SchedulingFilterStore.js]: Not used so delete :smile:

File: app/views/dashboard/index.haml.orig [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  %li{ class: project_type_class("potential") }
+=======
+  %li.active
+    = react_component('type-toggle')
+  %li.potential
+>>>>>>> Stashed changes
+    = link_to potential_dashboard_index_path do
+      Potential
+  %li{ class: project_type_class("archived") }
+    = link_to archived_dashboard_index_path do
```
Comment (nekath) [inline app/views/dashboard/index.haml.orig]: merge conflict

**Comment by nekath** [comment]
Why this branch has 80 commits? Something went wrong on rebasing.

File: app/views/dashboard/index.haml.orig (hunk 8 above)
Comment (smnkrt) [inline app/views/dashboard/index.haml.orig]: Thanks x1000! I had no idea i missed something like that.

**Comment by azranel** [comment]
Code LGTM :+1:. Im just not sure about those number of commits.

**Pull request merged**